sponsors/models.py

```diff
@@ -310,10 +316,16 @@ def estimated_cost(self):
         )
 
     def reject(self):
+        if self.REJECTED not in self.next_status:
+            msg = f"Can't reject a {self.get_status_display()} sponsorship."
+            raise SponsorshipInvalidStatusException(msg)
         self.status = self.REJECTED
         self.rejected_on = timezone.now().date()
 
     def approve(self):
+        if self.APPROVED not in self.next_status:
```

Comment:
we probably want to ensure that a valid start/end date was provided as part of the approval process. this should be enforced, as we'll need it for generating Statement of Work.

Reply:
Totally! I'll update this method to do so, but will do in the statement of work's branch, is it ok? I've changed how the accept view works over there, so it'll be better to change this from a more up to date code.

Reply:
Makes sense! Thank you.